src/sentry/static/sentry/app/views/organizationIntegrations/integrationRow.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import capitalize from 'lodash/capitalize';
+
+import Link from 'app/components/links/link';
+import {PanelItem} from 'app/components/panels';
+import PluginIcon from 'app/plugins/components/pluginIcon';
+import space from 'app/styles/space';
+import {
+  IntegrationProvider,
+  Integration,
+  PluginWithProjectList,
+  Organization,
+  PluginProjectItem,
+  SentryApp,
+  SentryAppInstallation,
+} from 'app/types';
+import {t} from 'app/locale';
+
+import IntegrationStatus from './integrationStatus';
+import {mapped} from './integrationUtil';
+
+type AppOrProviderOrPlugin = SentryApp | IntegrationProvider | PluginWithProjectList;
+
+type InstalledList = Integration[] | PluginProjectItem[] | SentryAppInstallation[];
+
+type Props = {
+  integration: AppOrProviderOrPlugin;
```

Comment:
@NisanthanNanthakumar I am against pushing this logic down to the child component. We can pass the `type` so can construct the `baseUrl` but I would much prefer not having to props here have such complicated types that are ORing three different types. We need that typing in `integrationListDirectory.tsx` but at this level, let's keep it more simple. It will make unit tests easier to write because you won't' have to write a case for every single integration type for every feature that needs testing. Here's what I think the types should be:

- counfigCount - number of configurations
- slug - slug of integration or plugin
- displayName - name we display which would include (Legacy) for deprecated plugins like Jira, Github, etc
- organization
- integrationType (only used in the URL)

Let's talk in person if you have any more questions...